autofit/graphical/factor_graphs/factor.py

```diff
@@ -273,12 +274,14 @@ def _function_shape(
         """
         Calculates the expected function shape based on the variables
         """
-        var_shapes = {
-            k: np.shape(x) for k, x in kwargs.items()}
+        var_shapes = {k: np.shape(x) for k, x in kwargs.items()}
         return self._var_shape(**var_shapes)
 
     @lru_cache()
-    def _var_shape(self, **kwargs: Tuple[int, ...]) -> Tuple[int, ...]:
+    def _plate_sizes(
+        self, **kwargs: Tuple[int, ...]
+    ) -> Tuple[int, Dict[Union[None, str], Tuple[int, ...]]]:
```

Comment:
That is a hell of a type annotation. Consider creating a class to comprise these variables

Reply:
That could be potentially worth it. I think we probably need to have a discussion about Plate notation in general, as I agree there's a lot of semi-duplicated functionality. `Tuple[int, Dict[Union[None, str], Int]]`

The key point of Plate functionality is to effectively keep track of NumPy dimensions.